tests/unit/linalg/test_matrix_hypre.cpp

```diff
@@ -100,6 +100,7 @@ TEST_CASE("HypreParMatrixAbsMult",  "[Parallel], [HypreParMatrixAbsMult]")
       X0.Randomize(1);
       X1.Randomize(1);
       A->AbsMultTranspose(3.4,Y0,-2.3,X0);
+      Aabs->EnsureMultTranspose();
```

Comment:
Why can't we put the Ensure call in the MultTranspose method?

Reply:
That is an option. I was also thinking if we should do that and if there are any big drawbacks to doing that.

This method (`EnsureMultTranspose`) was introduced in #3052, to be similar to the method with the same name in class `SparseMatrix`.

The reasoning for asking the user to explicitly call `EnsureMultTranspose`, if they need the action of the transpose, was to make it more obvious that `SparseMatrix` and `HypreParMatrix` construct and store the transpose in certain cases (e.g. when running on GPU). This was to make it more obvious to users that if they modify the matrix, then they will need to rebuild the transpose too. In the case of `SparseMatrix` this can be done by calling the method `ResetTranspose` whereas for `HypreParMatrix` we did not actually add such method, partially because we rarely, if ever, modify `HypreParMatrix` objects after constructing them.

So we can choose to hide `EnsureMultTranspose` inside `MultTranspose` (and other similar methods like `AddMultTraspose`), however that makes the fact that one needs to delete/rebuild the transpose when modifying the matrix less obvious.

Reply:
I see the dilemma.  I think it's best to require it before MultTranspose and put something in the MultTranspose documentation that says you must call EnsureMultTranspose to get it there and ResetTranspose if it is changed.

Reply:
@acfisher, @tzanio, I made the updates as we discussed offline. Take a look and let me know if I missed something.